Gifski/TimeRemainingEstimator.swift

```diff
@@ -10,13 +10,21 @@ final class TimeRemainingEstimator {
 	/// Begin fade out when remaining time reaches this amount.
 	let fadeOutThreshold: TimeInterval = 1
 
-	var progress: Progress? {
+	weak var progress: Progress? {
 		didSet {
-			progressObserver = progress?.observe(\.fractionCompleted) { sender, _ in
+			progressObserver = progress?.observe(\.fractionCompleted) { [weak self] sender, _ in
```

Comment:
Xcode complains that the observers leak, but I don't know any way to `nil` them when `progress` becomes nil from being weak, not from being set to `nil`. There's no way to observe when a weak variable becomes `nil`...

Reply:
why do you only want to know when it becomes `nil` due to weak referencing, not to `nil`ing it manually?

Reply:
I want both, but in this case, only knowing when it becomes nil through weak would be useful as we never manually `nil` it. However, it doesn't really matter in this case as `TimeRemainingEstimator` is released when the conversion is done anyway.